packages/valory/skills/decision_maker_abci/handlers.py

```diff
@@ -337,3 +347,19 @@ def _send_not_found_response(
         # Send response
         self.context.logger.info("Responding with: {}".format(http_response))
         self.context.outbox.put_message(message=http_response)
+
+    def _check_required_funds(self) -> bool:
+        """Check the agent has enough funds."""
+        return (
+            self.synchronized_data.wallet_balance
+            > self.context.params.agent_balance_threshold
+        )
+
+    def _check_is_receiving_mech_responses(self) -> bool:
+        """Check the agent is making on chain transactions."""
+        # Checks the most recent decision receive timestamp, which can only be returned after making a mech call
+        # (an on chain transaction)
+        return (
+            self.synchronized_data.decision_receive_timestamp
+            < 2 * self.context.params.reset_pause_duration
```

Comment:
Why compared with the reset and pause duration? Here we are saying:

if the mech response takes longer than twice the **duration** of the reset and pause round then we are not receiving mech responses. IMO we should introduce a separate param and set it to the expected mech time response.

Reply:
Should be done now.